From 176016647e2dede78a4407945cc95f96327e6e49 Mon Sep 17 00:00:00 2001
From: Viacheslav Bocharov <adeep@lexina.in>
Date: Wed, 29 May 2024 14:54:01 +0300
Subject: [PATCH 22/22] disable nor

---
 cmd/storage.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmd/storage.c b/cmd/storage.c
index 2eb17b65070..1e891a74857 100644
--- a/cmd/storage.c
+++ b/cmd/storage.c
@@ -25,7 +25,7 @@
 #undef pr_info
 #define pr_info       printf
 
-#ifdef CONFIG_SPI_FLASH
+#ifdef CONFIG_SPI_FLASH0
 extern int spi_nor_pre(void);
 extern int spi_nor_probe(u32 init_flag);
 #endif
@@ -77,7 +77,7 @@ static struct device_node_t device_list[] = {
 #ifdef CONFIG_MTD_SPI_NAND
 	{BOOT_SNAND, "spi-nand", spi_nand_pre, spi_nand_probe},
 #endif
-#if CONFIG_SPI_FLASH
+#if CONFIG_SPI_FLASH0
 	{BOOT_SNOR, "spi-nor", spi_nor_pre, spi_nor_probe},
 #endif
 #if 0
-- 
2.45.1

