From 1c8da1cc670e48370e3497f72147bd231c7ae265 Mon Sep 17 00:00:00 2001
From: Frank Wunderlich <frank-w@public-files.de>
Date: Mon, 14 Apr 2025 16:12:03 +0200
Subject: [PATCH 86/93] phy: mtk: disable powerdomain part in 2g5 driver

---
 drivers/net/phy/mediatek-2p5ge.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/phy/mediatek-2p5ge.c b/drivers/net/phy/mediatek-2p5ge.c
index d4e8365a076c..a01ebdc925ca 100644
--- a/drivers/net/phy/mediatek-2p5ge.c
+++ b/drivers/net/phy/mediatek-2p5ge.c
@@ -98,7 +98,7 @@ static int mt7988_2p5ge_phy_config_init(struct phy_device *phydev)
 	struct pinctrl *pinctrl;
 
 	if (!phy_priv->fw_loaded) {
-		pm_dev = dev_pm_domain_attach_by_name(dev, "i2p5gbe-pd");
+		/*pm_dev = dev_pm_domain_attach_by_name(dev, "i2p5gbe-pd");
 		if (IS_ERR(pm_dev)) {
 			ret = PTR_ERR(pm_dev);
 			dev_err(dev, "failed to get i2p5g pm-domain: %d\n", ret);
@@ -110,7 +110,7 @@ static int mt7988_2p5ge_phy_config_init(struct phy_device *phydev)
 		/* We need to add 1 to power domain counter first so that
 		 * we can correctly power off internal 2.5Gphy
 		 */
-		ret = pm_runtime_get_sync(pm_dev);
+		/*ret = pm_runtime_get_sync(pm_dev);
 		if (ret < 0) {
 			dev_err(&phydev->mdio.dev, "failed to power on!\n");
 			return ret;
@@ -120,7 +120,7 @@ static int mt7988_2p5ge_phy_config_init(struct phy_device *phydev)
 		if (ret < 0) {
 			dev_err(&phydev->mdio.dev, "failed to power on!\n");
 			return ret;
-		}
+		}*/
 
 		np = of_find_compatible_node(NULL, NULL, "mediatek,2p5gphy-fw");
 		if (!np)
-- 
2.49.0

